### PR TITLE
added function to clean meshes (remove duplicate vertices, faces)

### PR DESCRIPTION
**Pull request by jo-mueller**
Hi @haesleinhuepf ,

this fixes the issue with the smoothing you were describing in PR #23 . The problem here was a large number of duplicate vertices/faces in the mesh, which apparently happens easily with marching cubes outputs and has also been reported [here](https://github.com/marcomusy/vedo/issues/710) (other problem, same cause).

The `mesh.clean()` function [from vedo ](https://vedo.embl.es/autodocs/content/vedo/pointcloud.html#vedo.pointcloud.Points.clean) fixes this so I added it as a function. I also applied it to the output of the marching cubes so that downstream processing doesn't run into the same issue. Let me know what you think :)

### CODE REVIEW

**Comment by jo-mueller** [comment]
When I introduced the various subdivision methods in PR #25 , I noticed that all of the subdivision methods work on that branch - however, they don't work properly on this branch. It seems that `mesh.clean()` has some unexpected consequences for some other surface filters -  I guess this should thus be a separate function rather than built in marching cubes.

File: napari_process_points_and_surfaces/__init__.py [hunk 1]
```diff
@@ -492,7 +494,7 @@ def voxel_down_sample(points_data:PointsData, voxel_size: float = 5, viewer:napa
     new_point_cloud = point_cloud.voxel_down_sample(voxel_size)
 
     result = to_numpy(new_point_cloud.points)
-    return resul
+    return result
```
Comment (haesleinhuepf) [inline napari_process_points_and_surfaces/__init__.py:497]: Embarasseing 😅

File: napari_process_points_and_surfaces/__init__.py [hunk 2]
```diff
@@ -181,6 +182,7 @@ def label_to_surface(labels: LabelsData, label_id: int = 1) -> SurfaceData:
     label_id: int
     """
     from skimage.measure import marching_cubes
+    from ._vedo import vedo_remove_duplicate_vertices
```
Comment (haesleinhuepf) [inline napari_process_points_and_surfaces/__init__.py:185]:
```suggestion
```
Is this import necessary? I don't see where this function is used.

**Review by haesleinhuepf — APPROVED** [review]
Hi Johannes @jo-mueller ,

just one minor thing. I would merge afterwards...

Thanks!

Best,
Robert

File: napari_process_points_and_surfaces/__init__.py (hunk 1 above)
Comment (jo-mueller) [reply napari_process_points_and_surfaces/__init__.py:496]: Happens to all of us :D

**Comment by jo-mueller** [comment]
Hi Robert @haesleinhuepf ,

thanks for checking and spotting the unused import!

Best, Johannes

**Pull request merged**